metric_learn/base_metric.py

```diff
@@ -317,13 +331,16 @@ def predict(self, pairs):
     y_predicted : `numpy.ndarray` of floats, shape=(n_constraints,)
       The predicted learned metric value between samples in every pair.
     """
-    return self.decision_function(pairs)
+    check_is_fitted(self, ['threshold_', 'transformer_'])
+    return 2 * (self.decision_function(pairs) >= - self.threshold_) - 1
 
   def decision_function(self, pairs):
-    """Returns the learned metric between input pairs.
+    """Returns the decision function used to classify the pairs.
 
-    Returns the learned metric value between samples in every pair. It should
-    ideally be low for similar samples and high for dissimilar samples.
+    Returns the opposite of the learned metric value between samples in every
+    pair. Hence it should ideally be low for dissimilar samples and high for
```

Comment:
maybe add "to be consistent with scikit-learn convention"?

Reply:
Agreed, done